Client/Frontend/Notifications/NotificationRootViewController.swift

```diff
@@ -159,9 +161,34 @@ private extension NotificationRootViewController {
         }
     }
 
-    @objc func didFinishSyncing() {
-        showingNotification = false
-        dispatch_async(dispatch_get_main_queue()) {
+    @objc func didFinishSyncing(notification: NSNotification) {
+        let syncMessage: NSAttributedString?
+        if let syncManager = notification.object as? SyncManager,
+            let syncDisplayState = syncManager.syncDisplayState {
+            switch syncDisplayState {
+            case .Bad(let message):
+                syncMessage = NSAttributedString(string: message, attributes: [NSForegroundColorAttributeName: UIConstants.TableViewRowErrorTextColor, NSFontAttributeName: notificationView.titleLabel.font])
+            case .Stale(let message):
+                syncMessage = NSAttributedString(string: message, attributes: [NSForegroundColorAttributeName: UIConstants.TableViewRowWarningTextColor, NSFontAttributeName: notificationView.titleLabel.font])
+            default: syncMessage = nil
+            }
+        } else {
```

Comment:
If only. It actually complains that `syncMessage` isn't guaranteed to be initialized if I don't always provide it with a value, even if that value is nil. I think it's a result of it being a `let` rather than a `var`. Swift still has some annoying oddities.